docs/source/platform/schema-validation.md

```diff
@@ -1,60 +1,101 @@
+
+<div style="text-align:center">
+![multiple service checks](../img/schema-validation/service-checks.png)
+</div>
+
+<h2 id="categorize">Categorizing schema changes</h2>
+
+`apollo` buckets schema changes by impact on consumers. Since consumers choose exactly how to use the GraphQL API, the real effect of changes can be unpredictable when inspecting the change in isolation. To properly categorize these changes, the Apollo Platform matches changes with field usage metrics to determine a change's severity.
+
+<h3 id="severity">Change severity</h3>
+
+The Apollo Platform identifies three change severities and reports them on the command line or within a pull-request status([setup for GitHub](#github)):
+
+1. **Failure**: Either the schema is invalid or the changes _will_ break current clients.
+2. **Warning**: There are potential problems that may come from this change, but no clients are immediately impacted.
+3. **Notice**: This change is safe and will not break current clients.
+
+The overall status of validation, which modifies the cli's exit code and GitHub status, depends on whether metrics are reported to the service.
+
+<h3 id="change-types">Change types</h3>
+
+The Apollo Platform identifies three types of changes:
 
-Developers who have worked with REST APIs in the past have probably recognized various patterns for versioning the API, commonly by using a different URI (e.g. `/api/v1`, `/api/v2`, etc.) or a query parameter (e.g. `?version=1`).  With this technique, an application can easily end up with many different API endpoints over time, and the question of _when_ an API can be deprecated can become problematic.
+- **Addition**: additive change that is transparent to clients, such as field addition
+- **Update**: in-place modification of field or type that could change query results, such as kind change or required argument addition
+- **Removal**: removal of schema element that consumer use, such as type, field, enum, argument, or other
 
-It might be tempting to version a GraphQL API the same way, but it's unnecessary with the right techniques.  By following the strategies and precautions outlined in this guide and using Apollo tooling that adds clarity to every change, many iterations of an API can be served from a single endpoint.
+Each of these overall types contains different changes codes, such as `TYPE_REMOVED`, `ARG_DEFAULT_VALUE_CHANGE`, `ENUM_VALUE_ADDED`, or `FIELD_CHANGED_KIND`, which are displayed in the cli output.
 
-<h3 id="field-usage">Field usage</h3>
+<h3 id="metrics">Assigning severity</h3>
 
-Rather than returning extensive amounts of data which might not be necessary, GraphQL allows consumers to specify exactly what data they need.  This field-based granularity is valuable and avoids "over-fetching" but also makes it more difficult to understand what data is currently being used.
+When metrics are sent to a service, the changes are verified in the following manner:
```

Comment:
It might be more useful to walk through the algorithm. 
First: all removals are marked as failures and all updates are marked as warnings
Then: failures and warnings are compares to usage metrics. Failures are downgraded to notices if the fields are unused (maybe this should be warning?) and warnings are upgraded to failures if the modified fields / types have been used within the validation window.